packages/atomic/src/components/facets-v1/atomic-color-facet/atomic-color-facet.tsx

```diff
@@ -0,0 +1,328 @@
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+    const options: FacetOptions = {
+      facetId: this.facetId,
+      field: this.field,
+      numberOfValues: this.numberOfValues,
+      sortCriteria: this.sortCriteria,
+      facetSearch: {numberOfValues: this.numberOfValues},
+    };
+    this.facet = buildFacet(this.bindings.engine, {options});
+    this.facetId = this.facet.state.facetId;
+    this.bindings.store.state.facets[this.facetId] = {
+      label: this.label,
+    };
+  }
+
+  public componentShouldUpdate(
+    next: unknown,
+    prev: unknown,
+    propName: keyof AtomicColorFacet
+  ) {
+    if (propName === 'facetState') {
+      return shouldUpdateFacetSearchComponent(
+        (next as FacetState).facetSearch,
+        (prev as FacetState).facetSearch
+      );
+    }
+
+    return true;
+  }
+
+  private get numberOfSelectedValues() {
+    return this.facetState.values.filter(({state}) => state === 'selected')
+      .length;
+  }
+
+  private renderHeader() {
+    return (
+      <FacetHeader
+        i18n={this.bindings.i18n}
+        label={this.label}
+        onClearFilters={() => this.facet.deselectAll()}
+        numberOfSelectedValues={this.numberOfSelectedValues}
+        isCollapsed={this.isCollapsed}
+        onToggleCollapse={() => (this.isCollapsed = !this.isCollapsed)}
+      ></FacetHeader>
+    );
+  }
+
+  private renderSearchInput() {
+    const shouldDisplaySearch =
+      this.withSearch && this.facetState.canShowMoreValues;
+    if (!shouldDisplaySearch) {
+      return;
+    }
+
+    return (
+      <FacetSearchInput
+        i18n={this.bindings.i18n}
+        label={this.label}
+        query={this.facetState.facetSearch.query}
+        onChange={(value) => {
+          if (value === '') {
+            this.facet.facetSearch.clear();
+            return;
+          }
+          this.facet.facetSearch.updateText(value);
+          this.facet.facetSearch.search();
+        }}
+        onClear={() => this.facet.facetSearch.clear()}
+      ></FacetSearchInput>
+    );
+  }
+
+  private renderValue(facetValue: FacetValue, onClick: () => void) {
+    const displayValue = this.bindings.i18n.t(facetValue.value);
+    const isSelected = facetValue.state === 'selected';
+    switch (this.displayValuesAs) {
+      case 'checkbox':
+        return (
+          <FacetValueCheckbox
+            displayValue={displayValue}
+            numberOfResults={facetValue.numberOfResults}
+            isSelected={isSelected}
+            i18n={this.bindings.i18n}
+            onClick={onClick}
+            searchQuery={this.facetState.facetSearch.query}
+          >
+            <FacetValueLabelHighlight
+              displayValue={displayValue}
+              isSelected={isSelected}
+              searchQuery={this.facetState.facetSearch.query}
+            ></FacetValueLabelHighlight>
+          </FacetValueCheckbox>
+        );
+      case 'box':
+        return (
+          <FacetValueBox
```

Comment:
Same for this component, would it be enough to use the same or a new component is necessary?

Reply:
At first, I thought of adding classes to change the background color and other little style changes but it might be better to create different components.